transport/http/handler.go

```diff
@@ -82,6 +87,9 @@ func (h handler) callHandler(w http.ResponseWriter, req *http.Request) error {
 
 	ctx := context.Background()
 
+	ctx, span := h.createSpan(ctx, req, treq, start)
+	defer span.Finish()
```

Comment:
I still think this should be done after `ParseTTL` so that the context you're operating on has a timeout (which is something you can imagine being tagging the span with in the future).